doc/contribute/coding_guidelines/index.rst

```diff
@@ -944,3 +944,95 @@ Related GitHub Issues and Pull Requests are tagged with the `Inclusive Language
 .. _Bluetooth Appropriate Language Mapping Tables: https://btprodspecificationrefs.blob.core.windows.net/language-mapping/Appropriate_Language_Mapping_Table.pdf
 .. _OSHWA Resolution to Redefine SPI Signal Names: https://www.oshwa.org/a-resolution-to-redefine-spi-signal-names/
 .. _Zephyr issue 27033: https://github.com/zephyrproject-rtos/zephyr/issues/27033
+
+Parasoft Codescan Tool
+**********************
+
+Parasoft Codescan is an official static code analysis tool used by the Zephyr
+project. It is used to automate compliance with a range of coding and security
+standards.
+Currently, the tool is set to MISRA C:2012 Coding Standard because Zephyr
+:ref:`coding_guidelines` are based on that standard.
+It is used together with the Coverity Scan tool to achieve the best code health
+and precision in bug findings.
+
+Violations fixing process
+=========================
+
+Step 1
+  Any Zephyr Project member, company or a developer can request a license
+  to access Codescan online dashboard.
+
+Step 2
+  A developer starts to review raised violations.
+
+Step 3
+  A developer submits a Github PR with the fix. Commit messages should follow
+  the same guidelines as other PRs in the Zephyr project. Please add a comment
+  that your fix was found by a static coding scanning tool.
+  A developer should follow and refer to the Zephyr :ref:`coding_guidelines`
+  as the basic coding rules. These rules are based on the MISRA C standard.
+
+  Below you can find an example of a recommended commit message::
+
+     lib: os: add braces to 'if' statements
+
+
+     An 'if' (expression) construct shall be followed by a compound statement.
+     Add braces to improve readability and maintainability.
+
+     Found as a coding guideline violation (Rule 15.6) by static
+     coding scanning tool.
+
+     Signed-off-by: Johnny Developer johnny.developer@company.com
+
+Step 4
+  If a violation is a false-positive developer must submit a PR with
+  a suppression tag and an explanation comment.
+
```

Comment:
IMHO the suppression should not used for false-positives but for deviations. Deviations are real violations that *we* deliberated decided to ignore for a reason (that should be explained in the comment). False positives should just be marked in the tool just like we do for coverity.

I don't like the idea of adding comments in the code due a tool problem.

Reply:
We will suppress false positives and deviations. All code false positive violations should be marked inside of the code to be ignored by the tool. All deviations should be marked in the code too with the explanation comment. We shouldn't rely on the tool, so standalone explanation comment can help. Even if we will remove tags, comments will stay in the code.